### PR TITLE
[#310] upgrade to latest golangci-lint and enable new linters

### LINKED ISSUE #310: Upgrade golangci-lint to the latest version and enable new linters

### PR DESCRIPTION
**Pull request by sahilm**
There's a degree of subjectivity to linting. Each commit represents an atomic decision and is open to change/discussion. Please refer to the commit in question if you'd like to discuss/change any decision.

Resolves #310

### CODE REVIEW

File: pkg/checksum/checker.go [hunk 1]
```diff
@@ -292,10 +292,10 @@ func (c *Checker) replaceChunk(chunk *table.Chunk) error {
 	c.recopyLock.Lock()
 	defer c.recopyLock.Unlock()
 
-	if _, err := dbconn.RetryableTransaction(context.TODO(), c.db, false, dbconn.NewDBConfig(), deleteStmt); err != nil {
+	if _, err := dbconn.RetryableTransaction(ctx, c.db, false, dbconn.NewDBConfig(), deleteStmt); err != nil {
```
Comment (sahilm) [inline pkg/checksum/checker.go:295]: Hope there's no unintended side effects here. I don't think this is a problem but want to highlight this change. It's to satisfy the `contextcheck` linter.

File: pkg/migration/runner_test.go [hunk 2]
```diff
@@ -2368,7 +2368,7 @@ func TestSkipDropAfterCutover(t *testing.T) {
 	var tableCount int
 	err = m.db.QueryRow(sql).Scan(&tableCount)
 	assert.NoError(t, err)
-	assert.Equal(t, tableCount, 1)
+	assert.Equal(t, 1, tableCount)
```
Comment (sahilm) [inline pkg/migration/runner_test.go:2371]: To fix [testifylint: expected-actual](https://github.com/Antonboom/testifylint?tab=readme-ov-file#expected-actual)

File: pkg/checksum/checker.go (hunk 1 above)
Comment (morgo) [reply pkg/checksum/checker.go:295]: It looks good to me. I think this is a code improvement.

**Review by morgo — APPROVED** [review]
All look good to me. No issues.

**Review by morgo — APPROVED** [review]

**Pull request merged**